modules/sratools/prefetch/main.nf

```diff
@@ -0,0 +1,38 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process SRATOOLS_PREFETCH {
+    tag "$meta.id"
+    label 'process_low'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:meta, publish_by_meta:['id']) }
+
+    conda (params.enable_conda ? "bioconda::sra-tools=2.11.0" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/sra-tools:2.11.0--pl5262h314213e_0"
+    } else {
+        container "quay.io/biocontainers/sra-tools:2.11.0--pl5262h314213e_0"
+    }
+
+    input:
+    val(meta)
```

Comment:
```suggestion
    tuple val(meta), val(id)
```

Reply:
In theory this module should work just off an `id` that doesn't necessarily have to have any sample information stored in `map`. I wonder if the safer way of coding this is to have a separate `id` in the channel? This way developers can provide an empy `meta` if they don't have any sample information.